Woodoku/src/Entity/WoodokuBoard.py

```diff
@@ -166,7 +169,7 @@ def add_shape(self, shape: WoodokuShape, x: int, y: int) -> None:
     def get_score(self) -> int:
         return self.__scoreAgent.get_score()
 
-    def __find_groups(self) -> Tuple[Dict[str, int], Set[Tuple[int, int]]]:
+    def __find_groups(self) -> Tuple[Dict[str, int], List[Tuple[int, int]]]:
```

Comment:
```suggestion
    def __find_groups(self) -> Tuple[Dict[str, int], Iterable[Tuple[int, int]]]:
```
Perhaps we should just return a Iterable, since all that is required is to loop over all instance

Reply:
In that way, we have to change the signiture of methods in `_WoodokuBoardRepresentation`.

Reply:
Not really, `isinstance(some_list, Iterable) ` returns `true`

Reply:
The current signiture for `add_blocks` is 
```python
def add_blocks(self, blocks_coord: List[Tuple[int, int]]) -> None:
```

Isn't that we need it to change to 
```python
def add_blocks(self, blocks_coord: Iterable[Tuple[int, int]]) -> None:
```

Reply:
Yeah, you are right. my bad, interface include all it's child class not the other way around